### PR TITLE
Add documentation to common gates

### PR DESCRIPTION
**Pull request by dabacon**
* Reorder so that single qubit gates come first.
* Group instances at bottom of file.

### CODE REVIEW

**Comment by dabacon** [comment]
Part of #1149

File: cirq/ops/common_gates.py [hunk 1]
```diff
@@ -808,3 +906,73 @@ def Ry(rads: float) -> YPowGate:
...
+
+# The Pauli Z gate.
+#   [[1, 0],
+#    [0, -1]]
+Z = ZPowGate()
+
+
+# The Hadmard gate.
+#   [[s, s],
+#    [-s, s]]
```
Comment (kevinsung) [inline cirq/ops/common_gates.py]: Minus sign in the wrong place.

File: cirq/ops/common_gates.py [hunk 2]
```diff
@@ -808,3 +906,73 @@ def Ry(rads: float) -> YPowGate:
...
+#    [0, i]]
+S = Z**0.5  # Clifford S gate, [[1, 0], [0, e]] where e = exp(i pi 0.5)
+
+
+# The T gate
+#   [[1, 0]
+#    [0, exp(i pi / 4)]]
+T = Z**0.25 # Clifford T gate, [[1, 0], [0, e]] where e = exp(i pi 0.25)
+
+# The controlled phase gate.
```
Comment (kevinsung) [inline cirq/ops/common_gates.py]: "controlled Z" is clearer, since "phase" could indicate something other than -1

File: cirq/ops/common_gates.py [hunk 3]
```diff
@@ -808,3 +906,73 @@ def Ry(rads: float) -> YPowGate:
...
+# The Clifford S gate
+#   [[1, 0],
+#    [0, i]]
+S = Z**0.5  # Clifford S gate, [[1, 0], [0, e]] where e = exp(i pi 0.5)
+
+
+# The T gate
+#   [[1, 0]
+#    [0, exp(i pi / 4)]]
+T = Z**0.25 # Clifford T gate, [[1, 0], [0, e]] where e = exp(i pi 0.25)
```
Comment (Strilanc) [inline cirq/ops/common_gates.py]: T is not Clifford; repeated doc

File: cirq/ops/common_gates.py [hunk 4]
```diff
@@ -808,3 +906,73 @@ def Ry(rads: float) -> YPowGate:
...
+#   [[s, s],
+#    [-s, s]]
+# where s = sqrt(0.5).
+H = HPowGate()
+
+
+# The Clifford S gate
+#   [[1, 0],
+#    [0, i]]
+S = Z**0.5  # Clifford S gate, [[1, 0], [0, e]] where e = exp(i pi 0.5)
```
Comment (Strilanc) [inline cirq/ops/common_gates.py]: Repeated doc

Comment (Strilanc) [reply cirq/ops/common_gates.py]: Done.

File: cirq/ops/common_gates.py (hunk 3 above)
Comment (Strilanc) [reply cirq/ops/common_gates.py]: Done.

File: cirq/ops/common_gates.py [hunk 5]
```diff
@@ -12,98 +12,58 @@
...
+    XPowGate()**t = XPowGate(exponent=t) is given by the matrix
+        [[g·c, -i·g·s],
+         [-i·g·s, g·c]]
+    where
+        c = cos(π·t/2)
+        s = sin(π·t/2)
+        g = exp(i·π·t/2).
+
+    Note in particular that this gate has a global phase factor of
+    e^{i·π·t/2} from the traditionally defined as rotation matrices
```
[the remark this answers is not in the document]
Comment (Strilanc) [reply cirq/ops/common_gates.py]: Done.

File: cirq/ops/common_gates.py [hunk 6]
```diff
@@ -12,98 +12,58 @@
...
+"""Quantum gates that are commonly used in the literature.
+
+This module creates Gate instances for the following gates:
+    X,Y,Z: Pauli gates.
+    H,S: Clifford gates.
+    T: A non-Clifford gate.
+    CZ: Controlled phase gate.
+    CNOT: Controlled not gate.
+    SWAP: the swap gate.
+    ISWAP a swap gate with a phase on the swapped subspace.
```
[the remark this answers is not in the document]
Comment (Strilanc) [reply cirq/ops/common_gates.py]: Done.

File: cirq/ops/common_gates.py (hunk 2 above)
Comment (Strilanc) [reply cirq/ops/common_gates.py]: Done.

File: cirq/ops/common_gates.py (hunk 1 above)
Comment (Strilanc) [reply cirq/ops/common_gates.py]: Done.

**Pull request merged**